### PR TITLE
Add support for pipeline templates and template configs

### PR DESCRIPTION
**Pull request by nevins-b**
This PR also adds dependency management with [dep](https://github.com/golang/dep)

### CODE REVIEW

**Comment by ethanfrogers** [comment]
@nevins-b sorry for the delayed review! i'll give it a review ASAP! would you mind updating the README to reflect these changes?

File: spinnaker/resource_pipeline.go [hunk 1]
```diff
@@ -14,16 +14,22 @@ func resourcePipeline() *schema.Resource {
...
 			"name": {
 				Type:     schema.TypeString,
+				ForceNew: true,
 				Required: true,
 			},
 			"pipeline": {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"pipeline_id": {
```
Comment (ethanfrogers) [inline spinnaker/resource_pipeline.go]: I'm not familiar with with this property. Does this link a pipeline to a parent template?

File: spinnaker/resource_pipeline_template.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+		Create: resourcePipelineTemplateCreate,
+		Read:   resourcePipelineTemplateRead,
+		Update: resourcePipelineTemplateUpdate,
+		Delete: resourcePipelineTemplateDelete,
+		//Exists: resourcePipelineTemplateExists,
```
Comment (ethanfrogers) [inline spinnaker/resource_pipeline_template.go]: Did you want to leave this commented out? I noticed `resourcePipelineTemplateExists` is implemented below.

File: spinnaker/resource_pipeline.go (hunk 1 above)
Comment (nevins-b) [reply spinnaker/resource_pipeline.go]: It doesn't, this is a unique UUID that identifies the pipeline in spinnaker's backend. It's only needed for the UpdatePipeline call I believe

**Pull request merged**